onadata/libs/utils/csv_import.py

```diff
@@ -251,30 +273,54 @@ def submit_csv(username, xform, csv_file, overwrite=False):
     xl_datetime_columns = [
         dt.get('name') for dt in x_json.get('children')
         if dt.get('type') in XLS_DATETIME_FIELDS]
+    xl_integer_columns = [
+        dt.get('name') for dt in x_json.get('children')
+        if dt.get('type') == 'integer'
+    ]
+    xl_decimal_columns = [
+        dt.get('name') for dt in x_json.get('children')
+        if dt.get('type') == 'decimal'
+    ]
 
     try:
         for row in csv_reader:
+            _check_datatype_constraint(
+                row, xl_integer_columns,
+                err_msg='Unknown integer format: ',
+                constraint_check=lambda x: int(x))
+            _check_datatype_constraint(
+                row, xl_decimal_columns,
+                err_msg='Unknown decimal format: ',
+                constraint_check=lambda x: float(x))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Lambda may not be necessary](https://app.codacy.com/app/Ona/onadata/pullRequest?prid=4516347)